### PR TITLE
Fix/site demos redirect

### PR DESCRIPTION
**Pull request by shtrom**

### CODE REVIEW

**Comment by shtrom** [comment]
Needs Learnosity/salt-states#1551 to work

File: src/includes/mapping.php [hunk 1]
```diff
@@ -0,0 +1,28 @@
+<?php
+
+$mapping = [
```
Comment (shtrom) [inline src/includes/mapping.php]: @TheRealEdDawson This bad boy.

**Comment by gonzalozawa** [comment]
@shtrom I am happy with the PHP7 bugfix.
Regarding the redirect changes, do we need this? As far as I can see, this would be just working at the moment for the live progress report and I don't see many more redirects needed in the near future. Does it worth the change?

**Comment by shtrom** [comment]
The main point of the PR is to bring in the functionality. It is overall very painful to have links from bookmarks and browsing history just lead to 404 with nothing useful but a link to the front page. This provides the workhorse to make it a bit better

The live progress report was just an example, because that's the one that triggered me. @TheRealEdDawson said he was could add the other redirections (or anyone who finds something amiss, really).

**Comment by mcgoughn** [comment]
Who's this helping though? Is this just an internal painpoint, and we need a little time to adjust, rather than something that's bothering our clients? Demos is an external facing site, it's core purpose isn't for an internal audience, so complicating the site as an outward facing code sample, largely to benefit internal users, isn't something I overly like.

**Comment by shtrom** [comment]
I think it benefits both internal and external users (using either history, bookmarks, or non-updated search).

But it definitely benefits me 😁.
-- 
Olivier Mehani <olivier.mehani@learnosity.com>
Sent from my mobile, please excuse my brevity

**Comment by richshupe** [comment]
i'm in favor.

**Comment by gonzalozawa** [comment]
@shtrom we still have this open PR that will be good to close. Is it possible to get the PHP7 fix released for the moment and park the redirect functionality? To be honest, we have released the demos revamp more than 6 months ago. Do you still think that we need to introduce this functionality now? :)

**Comment by shtrom** [comment]
It is still an occasional pain point, but yeah, I can split it.

**Comment by shtrom** [comment]
The PHP7 fix on its own is in #100

**Comment by shtrom** [comment]
@gonzalozawa we are looking at getting the infra-side fixes merged for this https://github.com/Learnosity/salt-states/pull/1551 do you think we could merge this?

As I mentionned before, this is still an occasional pain point for dev teams.

**Comment by gonzalozawa** [comment]
@shtrom this should be transparent to demos-site users so feel free to go ahead with the merge.

**Pull request merged**